### ISSUE TITLE
Support new `PlatformDispatcher.onError`

### ISSUE THREAD

**Issue by marandaneto** [getsentry/sentry-dart#833]
https://github.com/flutter/engine/pull/32078
https://docs.google.com/document/d/1HdGLKoVw_KWLrVzDNZ8Lk5nN-biHQzi_CS5mCkpiDwA/edit#heading=h.pub7jnop54q0

**Comment by ueman** [comment]
This probably requires a similar "hack" as this https://github.com/getsentry/sentry-dart/blob/8fd364e3d0bfcd320e2a4b2a3c5faa8e9c66c673/flutter/lib/src/sentry_asset_bundle.dart#L163-L178 in order to be forward and backward compatible.


```dart
typedef ErrorCallback = bool Function(Object exception, StackTrace stackTrace);

try {
  final errorCallback = (WidgetsBinding.instance!.platformDispatcher as dynamic).onError as ErrorCallback?;

  (WidgetsBinding.instance!.platformDispatcher as dynamic).onError = (Object exception, StackTrace stackTrace) {
    errorCallback?.call(exception, stackTrace);
    Sentry.captureException(exception, stackTrace: stackTrace);
  }
} on NoSuchMethodError catch (_) {
  // PlatformDispatcher.onError is not supported on this Flutter version
}
```

**Comment by marandaneto** [comment]
Yep, it needs some testing first to be sure we don't double-report anything.

**Comment by ueman** [comment]
Isn't that what the `DeduplicateEventProcesser` is for?

**Comment by marandaneto** [comment]
> Isn't that what the `DeduplicateEventProcesser` is for?

Yes, but `DeduplicateEventProcesser` is meant for the user's mistake, which has some perf. impact by watching the reference, etc...
If we can guarantee that nothing is reported twice, we can spare some bits.

**Comment by ueman** [comment]
It sounds like this error handler replaces the `runZoneGuarded` global error handler on Flutter, see this comment https://github.com/flutter/engine/pull/32078#discussion_r844347256

That would be nice as Zones increase the startup time and can cause weird behavior. So that's definitely something which needs to be tested.

**Comment by ueman** [comment]
This handler is now available as of Flutter 3.1.0. As suspected, it can replace the runZoneGuarded error handler. It would be nice, when given the correct configuration, no zone is created as that would mean less app startup impact, since zones come with small cost. 

If you want, you can take over my implementation from https://github.com/ueman/sentry-dart-tools/blob/main/sentry_flutter_plus/lib/src/integrations/on_error_integration.dart it just needs tests.
